src/compiler/types.ts

```diff
@@ -2734,6 +2736,52 @@ namespace ts {
         resolveModuleNames?(moduleNames: string[], containingFile: string): ResolvedModule[];
     }
 
+    /* @internal */
+    export const enum TransformFlags {
+        // Facts
+        // - Flags used to indicate that a node or subtree contains syntax that requires transformation.
+        TypeScript = 1 << 0,
+        ContainsTypeScript = 1 << 1,
+        Jsx = 1 << 2,
+        ContainsJsx = 1 << 3,
+        ES7 = 1 << 4,
+        ContainsES7 = 1 << 5,
+        ES6 = 1 << 6,
+        ContainsES6 = 1 << 7,
+
+        // Markers
+        // - Flags used to indicate that a subtree contains a specific transformation.
+        ContainsDecorators = 1 << 8,
+        ContainsPropertyInitializer = 1 << 9,
+        ContainsLexicalThis = 1 << 10,
+        ContainsCapturedLexicalThis = 1 << 11,
+        ContainsDefaultValueAssignments = 1 << 12,
+        ContainsParameterPropertyAssignments = 1 << 13,
+        ContainsSpreadElementExpression = 1 << 14,
+        ContainsComputedPropertyName = 1 << 15,
+
+        // Assertions

```

Comment:
Yes, and the example above shows how they are used. The bitmask of an assertion is merely a shortcut used to ensure that both the "this node is a" flag (e.g. `TypeScript`) and "the subtree containing this node has a" flag (e.g. `ContainsTypeScript`) are set on the node.